app/models/friendship.rb

```diff
@@ -1,4 +1,8 @@
 class Friendship < ApplicationRecord
   belongs_to :user
   belongs_to :friend, class_name: 'User'
+  belongs_to :inviter, class_name: 'User'
+
+  validates :user_id, uniqueness: { scope: :friend_id, message: 'You have already sent a request to this user' }
+  validates :inviter_id, presence: true
 end
```

Comment:
- [ ] You should create an additional low with reversed user-friend pair, this can be done with the function below.
```
def confirm_friend
  self.update_attributes(confirmed: true)
  Friendship.create!(friend_id: self.user_id,
                  user_id: self.friend_id,
                  confirmed: true)
end
```